--- aMule-2.3.1/src/DownloadQueue.cpp.old	2012-06-04 07:06:34.738748481 -0400
+++ aMule-2.3.1/src/DownloadQueue.cpp	2012-06-04 07:07:29.158748495 -0400
@@ -1436,19 +1436,23 @@
 
 bool CDownloadQueue::AddED2KLink( const CED2KLink* link, uint8 category, CECPacket *response )
 {
+	bool s;
 	switch ( link->GetKind() ) {
 		case CED2KLink::kFile:
 			return AddED2KLink( dynamic_cast<const CED2KFileLink*>( link ), category ,response);
 			
 		case CED2KLink::kServer:
-			return AddED2KLink( dynamic_cast<const CED2KServerLink*>( link ) );
+			s = AddED2KLink( dynamic_cast<const CED2KServerLink*>( link ) );break;
 			
 		case CED2KLink::kServerList:
-			return AddED2KLink( dynamic_cast<const CED2KServerListLink*>( link ) );
+			s =  AddED2KLink( dynamic_cast<const CED2KServerListLink*>( link ) );break;
 			
 		default:
 			return false;
 	}
+	if(s == true && response != NULL)
+		response->AddTag(CECTag(EC_TAG_STRING, wxTRANSLATE("Operation was successful.")));
+	return s;
 }
 
 
