--- a/configure	2019-11-01 18:15:33.000000000 +0300
+++ b/configure	2020-04-15 13:39:29.189001500 +0300
@@ -14877,9 +14877,9 @@
 	HAVE_IPV6_SKIP_EXTHDR_ARGS=4
 
 else
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: 3 args" >&5
-$as_echo "3 args" >&6; }
-	HAVE_IPV6_SKIP_EXTHDR_ARGS=3
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: 4 args" >&5
+$as_echo "4 args" >&6; }
+	HAVE_IPV6_SKIP_EXTHDR_ARGS=4
 
 fi
 
--- a/configure.ac	2019-11-01 18:13:11.000000000 +0300
+++ b/configure.ac	2020-04-15 13:46:43.868688200 +0300
@@ -329,8 +329,8 @@
 	AC_MSG_RESULT(4 args)
 	AC_SUBST(HAVE_IPV6_SKIP_EXTHDR_ARGS, 4)
 else
-	AC_MSG_RESULT(3 args)
-	AC_SUBST(HAVE_IPV6_SKIP_EXTHDR_ARGS, 3)
+	AC_MSG_RESULT(4 args)
+	AC_SUBST(HAVE_IPV6_SKIP_EXTHDR_ARGS, 4)
 fi
 
 AC_MSG_CHECKING([kernel source for bool checkentry function prototype])
