diff -ruN udev-106-bak/extras/volume_id//lib/volume_id.c udev-106/extras/volume_id//lib/volume_id.c
--- udev-106-bak/extras/volume_id//lib/volume_id.c	2017-12-13 20:03:57.431643460 +0800
+++ udev-106/extras/volume_id//lib/volume_id.c	2017-12-13 21:25:17.641642879 +0800
@@ -94,6 +94,49 @@
 	return 0;
 }
 
+struct exfat_super_block {
+	uint8_t		magic[3];
+	uint8_t		fs_type[8];
+	uint32_t	blocksize;
+	uint64_t	dblocks;
+	uint64_t	rblocks;
+	uint32_t	dummy1[17];
+	uint8_t	        dummy4;
+	uint8_t	 	uuid[4];
+	uint8_t		label[4];
+	uint32_t	dummy2[15];
+	uint8_t		fname[12];
+	uint32_t	dummy3[2];
+	uint64_t	icount;
+	uint64_t	ifree;
+	uint64_t	fdblocks;
+} PACKED;
+
+int volume_id_probe_exfat(struct volume_id *id, uint64_t off, uint64_t size)
+{
+	struct exfat_super_block *esb;
+
+	info("probing at offset 0x%llx", (unsigned long long) off);
+
+	esb = (struct exfat_super_block *) volume_id_get_buffer(id, off, 0x200);
+	if (esb == NULL){
+		return -1;
+	}
+
+	if (memcmp(esb->fs_type, "EXFAT", 5) != 0){
+		return -1;
+	}
+
+	volume_id_set_label_raw(id, esb->fname, 12);
+	volume_id_set_label_string(id, esb->fname, 12);
+	volume_id_set_uuid(id, esb->uuid, UUID_DOS);
+
+	volume_id_set_usage(id, VOLUME_ID_FILESYSTEM);
+	id->type = "exfat";
+
+	return 0;
+}
+
 int volume_id_probe_filesystem(struct volume_id *id, uint64_t off, uint64_t size)
 {
 	if (id == NULL)
@@ -102,6 +145,9 @@
 	info("probing at offset 0x%llx, size 0x%llx",
 	    (unsigned long long) off, (unsigned long long) size);
 
+	if (volume_id_probe_exfat(id, off, size) == 0)
+		goto found;
+
 	if (volume_id_probe_vfat(id, off, size) == 0)
 		goto found;
 
